src/support/ScrollBar.ts

```diff
@@ -99,83 +140,52 @@ export default class ScrollBar extends ThemeableBase<ScrollBarProperties> {
 		}
 	}
 
-	private _onSliderDragStart = (evt: MouseEvent & TouchEvent) => {
-		evt.preventDefault();
-		const { horizontal } = this.properties;
-		this._dragging = true;
-		this._dragPosition = getAbsolutePosition(evt, horizontal);
-	}
-
-	private _onSliderDrag = (evt: MouseEvent & TouchEvent) => {
-		const {
-			_dragging,
-			_dragPosition,
-			properties: {
-				horizontal,
-				onScroll
-			}
-		} = this;
-		if (_dragging) {
-			evt.preventDefault();
-			const delta = this._toRelative(getAbsolutePosition(evt, horizontal) - _dragPosition);
-			onScroll && onScroll(delta);
-			this._dragPosition = getAbsolutePosition(evt, horizontal);
-		}
-	}
-
-	private _onSliderDragStop = (evt: MouseEvent & TouchEvent) => {
-		evt.preventDefault();
-		this._dragging = false;
-	}
-
-	public onElementCreated(element: HTMLElement, key: string) {
-		this._onDomUpdate(element, key);
-	}
-
-	public onElementUpdated(element: HTMLElement, key: string) {
-		this._onDomUpdate(element, key);
-	}
-
 	render() {
+		const domSize = this._getDomSize();
 		const {
 			horizontal = false,
 			key = DEFAULT_KEY,
 			position,
 			size = 0,
 			sliderMin = 10,
-			sliderSize = 0,
-			visible: propsVisible
+			sliderSize = domSize,
+			visible: propsVisible,
+			onScroll
 		} = this.properties;
-		const renderPosition = String(this._fromRelative(position)) + 'px';
-		const relativeSliderSize = this._fromRelative(sliderSize);
-		const renderSliderSize = String(relativeSliderSize > sliderMin ? relativeSliderSize : sliderMin) + 'px';
+
+		let dragging = false;
+		if (onScroll) {
+			const dragResult = this.meta(Drag).get('slider');
+			const delta = horizontal ? dragResult.delta.x : dragResult.delta.y;
+			dragging = dragResult.isDragging;
+			delta && onScroll(toRelative(delta, size, domSize));
+		}
+
+		const renderPosition = String(fromRelative(position, size, domSize)) + 'px';
+		const absoluteSliderSize = fromRelative(sliderSize, size, domSize);
+		const renderSliderSize = String(absoluteSliderSize > sliderMin ? absoluteSliderSize : sliderMin) + 'px';
 		const visible = sliderSize >= size ? false : propsVisible !== undefined ? propsVisible : this._visible;
-		const styles = horizontal ? {
-			left: renderPosition,
-			width: renderSliderSize
-		} : {
-			top: renderPosition,
-			height: renderSliderSize
+		const styles = {
+			[ horizontal ? 'left' : 'top' ]: renderPosition,
+			[ horizontal ? 'width' : 'height' ]: renderSliderSize
 		};
 
 		return v('div', {
-			classes: this.classes(css.root, css.vertical, visible ? css.visible : css.invisible),
+			classes: this.classes(
+				css.root,
+				horizontal ? css.horizontal : css.vertical,
+				visible || dragging ? css.visible : css.invisible
+			),
 			key,
 
+			onclick: this._onclick,
```

Comment:
`onpointerdown` ?

Reply:
That wouldn't work/would be wrong.  `onpointerdown` is when the pointer is down, equivalent to `onmousedown` not `onclick`.  [There is no "pointer event" for "click"](https://www.w3.org/Submission/pointer-events/#extensions-to-the-element-interface).  Single taps are reported as `onclick`.